CMakeLists.txt

```diff
@@ -44,7 +44,7 @@ endif()
 option(BPLIB_BUILD_TEST_TOOLS "Whether or not to build the test programs as part of BPLib (standalone POSIX builds only)" ${BPLIB_DEFAULT_BUILD_TEST_TOOLS})
 option(BPLIB_INSTALL_LIBRARIES "Whether or not to install the libraries" ${BPLIB_STANDALONE_BUILD_MODE})
 option(BPLIB_USE_EXTERNAL_OSAL "Whether to use an external OSAL package, if OSAL is selected as OS layer" ${BPLIB_STANDALONE_BUILD_MODE})
-option(BPLIB_ENABLE_UNIT_TESTS "Whether to build unit tests (requires NASA OSAL and UT Assert)" ${BPLIB_DEFAULT_BUILD_UNIT_TESTS})
+option(BPLIB_ENABLE_UNIT_TESTS "Whether to build unit tests (requires NASA OSAL and UT Assert)" ${ENABLE_UNIT_TESTS})
```

Comment:
This is not ideal because ENABLE_UNIT_TESTS might not always be defined, and this will default to OFF/FALSE in that case.  The logic was intended to default to ON if prerequisites were there.

I looked into this issue - seems that the root cause is that the CFE build scripts only pass what is directly defined by the user.  So if one specifies "ENABLE_UNIT_TESTS=FALSE" this works as intended, but if it is left out completely, then it gets the "ON" value because CFS build uses OSAL.

My recommendation is to change the line above (line 38) to:

`elseif (BPLIB_OS_LAYER STREQUAL OSAL AND NOT IS_CFS_ARCH_BUILD)`

Rather than only `BPLIB_OS_LAYER STREQUAL OSAL` as it is now.

Reply:
I tried and It works.